src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java

```diff
@@ -1441,8 +1441,17 @@ public Boolean getBooleanProp(String propertyName) throws ConfigurationException
         try {
             return esapiPropertyManager.getBooleanProp(propertyName);
         } catch (ConfigurationException ex) {
+
             String property = properties.getProperty( propertyName );
             if ( property == null ) {
+                if (propertyName.startsWith("Logger.")) {
```

Comment:
@mickeyz07 -- IMO, this is a major implementation kludge to me. Generally, I was expecting in whatever static initializer that checks the value for the `Logger.LogPrefix` property, that any error / exception that would be thrown that you would just treat it as 'true' and then log something via `logSpecial`. If for some reason that approach turns out to be impossible, then the next best alternative would be to create a special dedicated getLogPrefix method. Another preferred alternative (something I've been meaning to do for a while now) is to create a new `Boolean getBooleanProp(String propName, boolean default)` and use that, and just pass in 'true' for the default, so if the specified property name is not a boolean, it defaults to the specified default value (which should be true for `Logger.LogPrefix.` I am also very confused why this is even written the way that it it; it appears that you could omit the entire:
```java
                if (propertyName.startsWith("Logger.")) {
```
and just went with the specific property name check that follows. (And why _this_ specific check? I would have expected `Logger.LogPrefix` here, since that's the new property. So why `Logger.LogEncodingRequired`? I'm confused.)

@jeremiahjstacey - I'd like your opinion on this, but I don't like this **_at all._** Can your think of any alternatives.

Reply:
I think I agree @kwwall  - Updates in this file will lead to additional scope that I think we'll be better off not trying to tackle in this effort.

@mickeyz07 - My understanding of what the desire here is to try to solve the problem at the lowest level possible.  I think I agree with the intent, but I think you can agree that it's not the cleanest or most maintainable due to current restrictions in the library's interfaces.
To facilitate a faster integration of this feedback, I would like to propose the following:
1.  Revert all changes to DefaultSecurityConfiguration.java
2. Update JavaLogFactory and SLF4JLogFactory implementations to default to TRUE, attempt to read from properties, and useLogSpecial on failure

```
boolean logPrefix = true;
try {
  logPrefix = ESAPI.securityConfiguration().getBooleanProp(LOG_PREFIX);
} catch (ConfigurationException ex) {
ESAPI.securityConfiguration().logSpecial("Failed to read Log Prefix configuration. Defaulting to enabled. ", ex);
}
```

It's not the best possible solution, but I think that's the best we can do in the current library implementation.

Reply:
Okay, my thoughts, exactly, and what you described was my preferred approach at resolving this.

@mickeyz07 - Please make the change that @jeremiahjstacey detailed above. Thanks.